rustls/src/crypto/signer.rs

```diff
@@ -134,3 +135,28 @@ impl CertifiedKey {
             .ok_or(Error::NoCertificatesPresented)
     }
 }
+
+#[cfg_attr(not(any(feature = "aws_lc_rs", feature = "ring")), allow(dead_code))]
+pub(crate) fn public_key_to_spki(
+    alg_id: &AlgorithmIdentifier,
+    public_key: &[u8],
```

Comment:
I was thinking about this as well -- I ended up not suggesting it because the `.as_ref()` calls don't seem like a big deal and making this abstract would probably end up with a bunch of monomorphization copies.

Reply:
> making this abstract would probably end up with a bunch of monomorphization copies.

That's fair and I thought about this downside but the set of public keys implementations is small and so is the function code. In either case I don't feel strongly enough to measure so I'm happy either way :-)

Reply:
Me too! @ctz gets to be tie breaker. 👍

Reply:
This is slightly nerd-sniping me into spending some time with godbolt and see if rustc has any de-monomorphisation optimisation pass (like you can do manually with an inner function). In any case I don't think the function is big enough to worry about.

Reply:
Ahh https://rustc-dev-guide.rust-lang.org/backend/monomorph.html#polymorphization (but probably wouldn't apply in this case)